scipy/stats/stats.py

```diff
@@ -3192,6 +3198,21 @@ def f_oneway(*args):
     alldata = np.concatenate(args)
     bign = len(alldata)
 
+    # Check if each the values within each group are constant
+    # And the common value in at least one group is different
+    # from that in another group - special cases
+    # Based on https://github.com/scipy/scipy/issues/11669
+    const_groups = True
+    for group in args:
+        if not all(x == group[0] for x in group):
+            const_groups = False
+            break
+    if const_groups:
+        if len(set(group[0] for group in args)) > 1:
+            return F_onewayResult(np.inf, 0)
```

Comment:
So the analogous behavior in `pearson` will throw a warning here, so I think we want to do the same here, maybe something like:
```
if const_groups:
    warnings.warn(F_onewayConstantInputWarning())
    if len(set(group[0] for group in args)) > 1:
...
```
that way the same warning is used in either case. From my perspective it seems a bit much to have separate custom warnings for the 2 different constant cases but not firm on the opinion. 



for ref the warning: 
https://github.com/scipy/scipy/commit/1acf46f508afa2c6d498e1001ca17e8ad98b46ef#diff-cc307f50f4eb0f9917b55fdf1dbb9b8cR3237

from the Pearson case.